### ISSUE TITLE
Update Kubo to new HTTP Delegated Routing

### ISSUE THREAD

**Issue by BigLep** [ipfs/kubo#9422]
## Done Criteria
Kubo is implementing https://github.com/ipfs/specs/pull/337 using the HTTP Delegated Routing version from https://github.com/ipfs/go-delegated-routing/issues/63

Specifically:
1. Add a new router type to the config, something like http or http-content-routing
2. Remove reframe router type until its issues are resolved

## Why Important
See motivation in https://github.com/ipfs/specs/pull/337 

## Notes
* Estimate in https://github.com/ipfs/go-delegated-routing/issues/63 was 2 developer days

**Comment by lidel** [comment]
Quick notes about default, implicit configuration (based on feedback from with Juan today):

- Change routing defaults in Kubo to DHT + implicit list of HTTP routers (https://github.com/ipfs/specs/pull/337)
  - Follow UX established for [DNS](https://github.com/ipfs/kubo/blob/master/docs/config.md#dns)
    -  user config remains empty, which allows us to update/expand implicit defaults in future releases
    - user is in control and can remove/update implicit routers if they wish.
  - only cid.conctact for now, we can add Cloudflare (already used for DoH) when they run their own indexer
- Probably a good time to switch [Routing.Type](https://github.com/ipfs/kubo/blob/master/docs/config.md#routing) to `OptionalString` 
  - write config migration that removes `Routing.Type` (switching to implicit default) when value is `dht`
  - change implicit default from `dht` to `auto` and document it means DHT + specific implicit routers

**Comment by BigLep** [comment]
I didn't see this earlier but I created https://github.com/ipfs/kubo/issues/9454 for tracking the work of adding cid.contact as an implicit default content router.  

I was imaging one PR to update the HTTP delegated routing client used by Kubo (for this issue) and then another PR for the cid.contact integration (including changelog/docs - which would be against #9454).  If it makes sense to do one large PR, please go ahead.

**Comment by ajnavarro** [comment]
Main integration here: https://github.com/ipfs/kubo/pull/9453

**Comment by BigLep** [comment]
@guseggert and @ajnavarro : what is remaining here before this is done?  Is it https://github.com/ipfs/kubo/pull/9475 ?

**Comment by BigLep** [comment]
2022-12-08: just waiting on https://github.com/ipfs/kubo/issues/9422 to be merged (as it also includes important docs items) and then this can be resolved.
